test/e2e/loadbalancer_plugin_test.go

```diff
@@ -0,0 +1,158 @@
+func setupLBPluginTest(testContext *e2e.TestContext, lbConfig *glooV1.LoadBalancerConfig) {
+	upstream := testContext.TestUpstream().Upstream
+	upstream.LoadBalancerConfig = lbConfig
+
+	customVS := helpers.NewVirtualServiceBuilder().
+		WithName("simple-lb").
+		WithNamespace(writeNamespace).
+		WithDomain("custom-domain.com").
+		WithRoutePrefixMatcher(e2e.DefaultRouteName, "/endpoint").
+		WithRouteActionToUpstream(e2e.DefaultRouteName, upstream).
+		Build()
+
+	testContext.ResourcesToCreate().VirtualServices = v1.VirtualServiceList{
+		customVS,
+	}
+}
+
+var _ = FDescribe("Load Balancer Plugin", Label(), func() {
+	var (
+		testContext *e2e.TestContext
+	)
+
+	BeforeEach(func() {
+		var testRequirements []testutils.Requirement
+
+		testContext = testContextFactory.NewTestContext(testRequirements...)
+		testContext.BeforeEach()
+	})
+
+	AfterEach(func() {
+		testContext.AfterEach()
+	})
+
+	JustBeforeEach(func() {
+		testContext.JustBeforeEach()
+	})
+
+	JustAfterEach(func() {
+		testContext.JustAfterEach()
+	})
+
+	Context("Maglev LoadBalancer", func() {
+		BeforeEach(func() {
+			setupLBPluginTest(testContext, &glooV1.LoadBalancerConfig{
+				Type: &glooV1.LoadBalancerConfig_Maglev_{
+					Maglev: &glooV1.LoadBalancerConfig_Maglev{},
+				},
+			})
+		})
+
+		It("can route traffic", func() {
+			requestBuilder := testContext.GetHttpRequestBuilder().
+				WithHost("custom-domain.com").
+				WithPath("endpoint")
+
+			Eventually(func(g Gomega) {
+				g.Expect(testutils.DefaultHttpClient.Do(requestBuilder.Build())).Should(matchers.HaveOkResponse())
+			}, "5s", ".5s").Should(Succeed())
+		})
+
+		It("should have expected envoy config", func() {
+			Eventually(func(g Gomega) {
+				dump, err := testContext.EnvoyInstance().StructuredConfigDump()
+				g.Expect(err).NotTo(HaveOccurred())
+
+				dacs, err := findDynamicActiveClusters(dump)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(dacs).NotTo(BeEmpty())
+				g.Expect(dacs).To(HaveLen(1))
+
+				g.Expect(dacs[0].LbPolicy).To(Equal(envoy_config_cluster_v3.Cluster_MAGLEV))
+				g.Expect(dacs[0].CommonLbConfig).To(BeNil())
+			}, "5s", ".5s").Should(Succeed())
+		})
+
+		XIt("does not close connections on host set change", func() {})
+	})
+
+	Context("Maglev LB w/ close connections on set change", func() {
+		BeforeEach(func() {
+			setupLBPluginTest(testContext, &glooV1.LoadBalancerConfig{
+				Type: &glooV1.LoadBalancerConfig_Maglev_{
+					Maglev: &glooV1.LoadBalancerConfig_Maglev{},
+				},
+				CloseConnectionsOnHostSetChange: true,
+			})
+		})
+
+		It("should have expected envoy config", func() {
+			Eventually(func(g Gomega) {
+				dump, err := testContext.EnvoyInstance().StructuredConfigDump()
+				g.Expect(err).NotTo(HaveOccurred())
+
+				dacs, err := findDynamicActiveClusters(dump)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(dacs).NotTo(BeEmpty())
+				g.Expect(dacs).To(HaveLen(1))
+
+				g.Expect(dacs[0].LbPolicy).To(Equal(envoy_config_cluster_v3.Cluster_MAGLEV))
```

Comment:
nit:
```suggestion
				g.Expect(dacs[0].LbPolicy).To(Equal(envoy_config_cluster_v3.Cluster_MAGLEV), dacs[0])
```
Might be a cleaner troubleshooting output

Reply:
Thank you. I will make sure to include more detailed assertion failure descriptions.